src/backend/distributed/commands/role.c

```diff
@@ -196,6 +196,12 @@ PreprocessAlterRoleSetStmt(Node *node, const char *queryString)
 	AlterRoleSetStmt *stmt = castNode(AlterRoleSetStmt, node);
 	ObjectAddress address = GetObjectAddressFromParseTree(node, false);
```

Comment:
Can you move the `if` statement above this line, so we don't calculate address unnecessarily.